src/CMakeLists.txt

```diff
@@ -51,28 +51,11 @@ if( WIN32 )
 	add_definitions( -D_WIN32 )
 
 
-	if( ( MSVC14 AND NOT CMAKE_GENERATOR_TOOLSET STREQUAL "v140_xp" ) OR	# For VS 2015.
-		( MSVC15 AND NOT CMAKE_GENERATOR_TOOLSET STREQUAL "v141_xp" ) )		# For VS 2017.
+	if( MSVC AND MSVC_VERSION GREATER 1900 )	# For VS 2017 and later.
```

Comment:
What is the case you want to handle with this version check? In other words, `if( MSVC )` seems to be enough.

Reply:
True, older versions are not actually supported. In my branch i support 2015.